test/intl402/DateTimeFormat/constructor-options-invalid-explicit-components.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-createdatetimeformat
+description: >
+  Verifies that constructor throws when dateStyle is combined with explicit format components.
+info: |
+    InitializeDateTimeFormat ( dateTimeFormat, locales, options )
+    ...
+    39. Let dateStyle be ? GetOption(options, "dateStyle", string, « "full", "long", "medium", "short" », undefined).
+    40. Set dateTimeFormat.[[DateStyle]] to dateStyle.
+    41. Let timeStyle be ? GetOption(options, "timeStyle", string, « "full", "long", "medium", "short" », undefined).
+    42. Set dateTimeFormat.[[TimeStyle]] to timeStyle.
+    43. If dateStyle is not undefined or timeStyle is not undefined, then
+
+        a. If hasExplicitFormatComponents is true, then
+            i. Throw a TypeError exception.
+        b. If required is date and timeStyle is not undefined, then
+            i. Throw a TypeError exception.
+        c. If required is time and dateStyle is not undefined, then
+            i. Throw a TypeError exception.
+        d. Let styles be dataLocaleData.[[styles]].[[<resolvedCalendar>]].
+        e. Let bestFormat be DateTimeStyleFormat(dateStyle, timeStyle, styles).
+---*/
+
+function optionsThrow(options, testName){
+  assert.throws(TypeError, function() {
+    new Intl.DateTimeFormat(undefined, options);
+  }, testName);
+}
+
+optionsThrow({dateStyle: "full", era: "long"}, "dateStyle-era: ");
```

Comment:
Why the trailing colon in the test name?

Reply:
The formatting on fail looks a little weird otherwise. for example:

```js
optionsThrow({timeStyle: "full"}, "timeStyle-should-not-throw");
```

``` 
FAIL constructor-options-invalid-explicit-components.js (default)
  timeStyle-should-not-throw Expected a TypeError to be thrown but no exception was thrown at all
  ```

I've changed this one to append the trailing colon in `optionsThrow` rather than including it in the test name.